### PR TITLE
adding ppl namespace api specs

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
adding ppl namespace specs

### Issues Resolved
[#231]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Tokesh** [comment]
Still a draft. I want to refactor the structures, as there is an intersection with the SQL plugin. But the main work is done and the tests are written.

File: spec/namespaces/ppl.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    ppl.get_stats::query.format:
+      name: format
+      in: query
+      description: A short version of the Accept header, e.g. json, yaml.
+      schema:
+        type: string
+    ppl.get_stats::query.sanitize:
+      name: sanitize
+      in: query
+      description: Specifies whether to escape special characters in the results
```
Comment (dblock) [inline spec/namespaces/ppl.yaml:105]: Let's add periods at the end of all these sentences.

File: spec/schemas/_plugins._common.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+openapi: 3.1.0
+info:
+  title: Schemas of _plugins._common category.
+  description: Schemas of _plugins._common category.
+  version: 1.0.0
+paths: {}
+components:
+  schemas:
+    Query:
```
Comment (dblock) [inline spec/schemas/_plugins._common.yaml:9]: Are these common to all plugins or just SQL? Feels like sql, so should it go to `_plugins/sql/_common.yaml`?

File: tests/ppl/explain.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - path: /books
+    method: DELETE
+    status: [200, 404]
+chapters:
+  - synopsis: Get explain of SQL Query
+    path: /_plugins/_ppl/_explain
+    method: POST
+    request_body:
+      payload:
+        query: search source=books;
```
Comment (dblock) [inline tests/ppl/explain.yaml:20]: Curious, does it require that semicolon?

File: tests/ppl/explain.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  - path: /books
+    method: PUT
+    request_body:
+      payload: {}
+epilogues:
+  - path: /books
+    method: DELETE
+    status: [200, 404]
+chapters:
+  - synopsis: Get explain of SQL Query
```
Comment (dblock) [inline tests/ppl/explain.yaml:15]: Add a period. We should really add it to the linter :)

**Review by dblock — CHANGES_REQUESTED** [review]
Nice. Mostly nitty asks.

File: tests/ppl/explain.yaml (hunk 4 above)
Comment (Tokesh) [reply tests/ppl/explain.yaml:15]: Yes, I will try to finish the linter, because for me it happens so often :)

File: tests/ppl/explain.yaml (hunk 3 above)
Comment (Tokesh) [reply tests/ppl/explain.yaml:20]: I just checked, it works without semicolons.
There are no semicolons in the examples of working with the OS API. That's why I removed them in the commit

File: spec/schemas/_plugins._common.yaml (hunk 2 above)
Comment (Tokesh) [reply spec/schemas/_plugins._common.yaml]: I got this, you're right! I fixed it

File: tests/ppl/explain.yaml (hunk 4 above)
Comment (dblock) [reply tests/ppl/explain.yaml:15]: I had a PR almost ready for some of it, https://github.com/opensearch-project/opensearch-api-specification/pull/463.

**Review by dblock — APPROVED** [review]

**Pull request merged**